pkg/java/pom/parse.go

```diff
@@ -371,7 +381,13 @@ func (p *parser) parseDependencies(deps []pomDependency, props map[string]string
 		if (d.Scope != "" && d.Scope != "compile") || d.Optional {
 			continue
 		}
-		dependencies = append(dependencies, d.ToArtifact(exclusions))
+
+		art := d.ToArtifact(opts.exclusions)
+		if !opts.lineNumber {
+			art.Locations = nil
+		}
```

Comment:
@knqyf263  what if we move this logic to `ToArtifact`.
we can add a location if needed instead of removing it.
something like this:
```go
func (d pomDependency) ToArtifact(opts analysisOptions) artifact {
	// To avoid shadow adding exclusions to top pom's,
	// we need to initialize a new map for each new artifact
	// See `exclusions in child` test for more information
	exclusions := map[string]struct{}{}
	if opts.exclusions != nil {
		exclusions = maps.Clone(opts.exclusions)
	}
	for _, e := range d.Exclusions.Exclusion {
		exclusions[fmt.Sprintf("%s:%s", e.GroupID, e.ArtifactID)] = struct{}{}
	}

	var locs types.Locations
	if opts.lineNumber {
		locs = types.Locations{
			{
				StartLine: d.StartLine,
				EndLine:   d.EndLine,
			},
		}
	}
	return artifact{
		GroupID:    d.GroupID,
		ArtifactID: d.ArtifactID,
		Version:    newVersion(d.Version),
		Exclusions: exclusions,
		Locations:  locs,
	}
}
```

Reply:
Yes, I also had the idea in my mind. Both work for me. You can decide it.

Reply:
Added changes in https://github.com/aquasecurity/go-dep-parser/pull/287/commits/e8ff04cc7449918aa92adce18250fa3455cb8f18